### ISSUE TITLE
Add IF NOT EXISTS support to CREATE TABLE AS

### ISSUE THREAD

**Issue by amotl** [crate/crate#14834]
### CrateDB version

nightly

### CrateDB setup information

```
docker run --rm -it --name=cratedb --publish=4200:4200 --publish=5432:5432     --env=CRATE_HEAP_SIZE=4g crate/crate:nightly
```

### Problem description

When using the `CREATE TABLE ... AS ...` statement, you can not use the `IF NOT EXISTS` clause.

### Steps to Reproduce

```sql
CREATE TABLE IF NOT EXISTS foobar AS SELECT 1;
```

### Actual Result

```
SQLParseException[line 1:36: mismatched input 'AS' expecting {'.', '('}]
```

### Expected Result

```
CREATE OK, 1 row affected  (0.091 sec)
```

### References

- https://github.com/crate-workbench/cratedb-toolkit/pull/56#pullrequestreview-1667420193

**Comment by sayyidhussain** [comment]
Hi Is this open to work i would like to work on this

**Comment by seut** [comment]
@sayyidhussain Sure, thanks!

**Comment by avnichauhan12** [comment]
Hii, Can i work on this.

**Comment by amotl** [comment]
Hi Avni. There is already a pull request at GH-14886, to start with a corresponding improvement. Let's see how that goes.

**Comment by dshunter107** [comment]
Has progress been made on this since the last discussion on GH-14886?

**Comment by dshunter107** [comment]
Hello @amotl , may I be assigned this?

**Comment by matriv** [comment]
@dshunter107 thx for volunteering.
Since it's been long time with no response, please go ahead.

**Comment by dshunter107** [comment]
hello, I am creating the tests in CreateTableAsAnalyzerTest. I wanted clarification on something. If the table already exists, should an error be thrown, saying that the table already exists, or should the program just do nothing?

As of now, I am going to assume that the program should just do nothing, but let me know if I am making the wrong assumption.

**Comment by matriv** [comment]
@amotl The feature has been implemented and will be available with the `5.8.0` release.
Thx!
